client/my-sites/stats/controller.jsx

```diff
@@ -226,12 +227,13 @@ export function site( context, next ) {
 }
 
 export function redirectToDaySummary( context ) {
-	// Query string from window.location.search differs depending on environment.
-	// Make sure to append the query if we are working inside wp-admin otherwise it will be lost.
 	const isWpAdmin = context.canonicalPath.includes( '/wp-admin/' );
-	const query =
-		isWpAdmin && context.query ? `&${ new URLSearchParams( context.query ).toString() }` : '';
-	const url = `/stats/day/${ context.params.module }/${ context.params.site }${ window.location.search }${ query }`;
+	const url = parseQueryStringForOdysseyRedirect(
+		`/stats/day/${ context.params.module }/${ context.params.site }${ window.location.search }`,
```

Comment:
We might not need `${ window.location.search }` here, which probably caused the issue of multiple `page=stats` issue.

I think we could use `query`, `path` and/or `pathname` from `context`, which are unified in both Odyssey and Calypso, which apparently I missed when adding the redirection logic.

Reply:
Looks great!

Reply:
It seems we only need the base pathname and append the `context.query` param string to it, which works on both Caylpso and Odyssey Stats. Updated in [0fc48b5](https://github.com/Automattic/wp-calypso/pull/97658/commits/0fc48b58f967c4968ff061e208bb3923e813bae0).